File: HibernateExample/src/main/java/com/revature/util/SessionFactoryUtil.java
package com.revature.util;

import org.hibernate.SessionFactory;
import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
import org.hibernate.cfg.Configuration;
import org.hibernate.service.ServiceRegistry;

public class SessionFactoryUtil {
	
	private static SessionFactory sf;
	
	static {
		Configuration configuration = new Configuration().configure();
		configuration.setProperty("hibernate.connection.username", System.getenv("BEARS_USERNAME"));
		configuration.setProperty("hibernate.connection.password", System.getenv("BEARS_PASSWORD"));
		configuration.setProperty("hibernate.connection.url", 
				"jdbc:oracle:thin:@db0715java.ce8jdxpwtsbf.us-east-2.rds.amazonaws.com:1521:orcl");
		//configuration.setProperty("hibernate.default_schema", System.getenv("BEARS_SCHEMA"));
		
		ServiceRegistry serviceRegistry = new StandardServiceRegistryBuilder()
				.applySettings(configuration.getProperties()).build();
		sf = configuration.buildSessionFactory(serviceRegistry);
	}
	
	public static SessionFactory getSessionFactory() {
		return sf;
	}

}

File: HibernateKotlin/src/main/kotlin/hibernate/kotlin/utility/SessionFactoryUtil.kt
package hibernate.kotlin.utility

import org.hibernate.SessionFactory
import org.hibernate.boot.registry.StandardServiceRegistryBuilder
import org.hibernate.cfg.Configuration
import org.hibernate.service.ServiceRegistry

object SessionFactoryUtil {

    var sessionFactory: SessionFactory

    init {
        val configuration = Configuration().configure()
        configuration.setProperty("hibernate.connection.username", System.getenv("DB_USERNAME"))
        configuration.setProperty("hibernate.connection.password", System.getenv("DB_PASSWORD"))
        configuration.setProperty("hibernate.connection.url", System.getenv("DB_URL"))
        configuration.setProperty("hibernate.default_schema", System.getenv("DB_SCHEMA"))
        val serviceRegistry = StandardServiceRegistryBuilder()
            .applySettings(configuration.properties).build()
        sessionFactory = configuration.buildSessionFactory(serviceRegistry)
    }
}
